<a href="https://colab.research.google.com/github/dote7ant/Moringa_DSPFT4/blob/master/ip.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**1. Get our libraries.**

In [0]:
# import our libraries
# get the pandas library 
import pandas as pd
# get the numpy library
import numpy as an

**2. Load our csv files.**

In [0]:
# load our files
url = "/content/Telcom_dataset.csv"
url1 = "/content/Telcom_dataset2.csv"
url2 = "/content/Telcom_dataset3.csv"
url3 = "/content/cells_geo.csv"
url4 = "/content/cells_geo_description.xlsx"
url5 = "/content/CDR_description.xlsx"

**3. Load our files to our data sets and preview them.**

In [71]:
# load our data sets in pandas
# this is data for may 6 23:04:37 to may 7 00:01:54
df = pd.read_csv(url)
df.head()

PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [52]:
# this is data for may 7 23:02:28 to may 8 00:01:53
df1 = pd.read_csv(url1)
df1.head(10)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO
5     sms      0  2012-05-07 23:05:41.0  ...     19e2e  /MMplBlWjh  /MMplBlWjh
6   Voice     41  2012-05-07 23:06:14.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
7   Voice     11  2012-05-07 23:06:28.0  ...     19e2e  +gySf02tKW  +gySf02tKW
8     sms      0  2012-05-07 23:06:35.0  ...     OTHER  /0FxP3az1x  /0FxP3az1x
9     sms      0  2012-05-07 23:06:50.0  ...     19e2e  0ecjmwPX21  0ecjmwPX21

[10 rows x 10 columns]

In [124]:
# this is data for may 8 23:01:28 to may 00:01:43
df2 = pd.read_csv(url2)
df2.head()

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[5 rows x 10 columns]

In [0]:
#columns = ['Villes', 'Status', 'Localisation', 'DecoupZone', 'Zonename', 'Longitude', 'Latitude', 'Region', 'Area', 'Cell_ID', 'Site_Code']
# this shows our cells and where they are
df3 = pd.read_csv(url3, delimiter=";")
df3.head()



Unnamed: 0   VILLES      STATUS  ... AREA     CELL_ID   SITE_CODE
0           0   ADJAME  In Service  ...  NaN  fbb2e75af3  0bb492ef17
1           1   ADJAME  In Service  ...  NaN  1502501d78  0bb492ef17
2           2   ADJAME  In Service  ...  NaN  0c2e182a0c  0bb492ef17
3           3  ASSINIE  In Service  ...  NaN  69bdfc69fd  8a1e6ce31f
4           4  ASSINIE  In Service  ...  NaN  7a1029d2ab  8a1e6ce31f

[5 rows x 12 columns]

In [53]:
# shows what the columns in cells_geo.csv mean
df4 = pd.read_excel(url4)
df4

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [54]:
# shows what the column names in the telecom data mean
df5 = pd.read_excel(url5)
df5

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

**4. TELECOM Data Cleanup**

In [72]:
# start will telecom data set 1
# find all null values
print(df.isnull().sum())

PRODUTC              0
VALUE                0
DATETIME             0
CELL_ON_SITE         0
DW_A_NUMBER_INT      0
DW_B_NUMBER_INT      0
COUNTRY_A            0
COUNTRY_B            0
CELL_ID              0
SITE_ID            716
dtype: int64


 We note that SITE_ID has 716 cells that have no value. We then do the following:



1.   Rename the "PRODUTC" column to 'PRODUCT' column.
2.    Drop CELL_ON_SITE, COUNTRY_A and COUNTRY_B as we have been informed they are not needed in the CDR_Description data.



In [73]:
# Rename the produtc column to product
df.columns = ['PRODUCT', 'VALUE', 'DATETIME', 'CELL_ON_SITE', 'DW_A_NUMBER_INT', 'DW_B_NUMBER_INT', 'COUNTRY_A', 'COUNTRY_B', 'CELL_ID', 'SITE_ID']
df.head()

PRODUCT  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [74]:
# drop the column CELL_ON_SITE,
df.drop('CELL_ON_SITE', axis=1, inplace=True)
df

# Drop the column COUNTRY_A
df.drop('COUNTRY_A', axis=1, inplace=True)
df

# Drop the column COUNTRY_B
df.drop('COUNTRY_B', axis=1, inplace=True)
df

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0      Voice      0  ...  /TJNe+Mmtu  /TJNe+Mmtu
1        sms      0  ...  /+cKZKIp41  /+cKZKIp41
2        sms     25  ...  /u0FSD+ahi  /u0FSD+ahi
3      Voice     56  ...  +cNeJzsTp3  +cNeJzsTp3
4      Voice     14  ...  051KhYDCpv  051KhYDCpv
...      ...    ...  ...         ...         ...
4996     sms    100  ...  0c2371d9bc  1e751d08a4
4997   Voice      1  ...  5fb791cf56  65efe2bc62
4998   Voice      0  ...  7773627c12  9ef9616350
4999   Voice      0  ...  3f4af49d6c  969b8d0e39
5000   Voice      7  ...  6f4d9b77a5  ebfc58af70

[5001 rows x 7 columns]

In [77]:
# data clean up 
# drop NA missing values 
clean_df = df.dropna()
clean_df

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0      Voice      0  ...  /TJNe+Mmtu  /TJNe+Mmtu
1        sms      0  ...  /+cKZKIp41  /+cKZKIp41
2        sms     25  ...  /u0FSD+ahi  /u0FSD+ahi
3      Voice     56  ...  +cNeJzsTp3  +cNeJzsTp3
4      Voice     14  ...  051KhYDCpv  051KhYDCpv
...      ...    ...  ...         ...         ...
4996     sms    100  ...  0c2371d9bc  1e751d08a4
4997   Voice      1  ...  5fb791cf56  65efe2bc62
4998   Voice      0  ...  7773627c12  9ef9616350
4999   Voice      0  ...  3f4af49d6c  969b8d0e39
5000   Voice      7  ...  6f4d9b77a5  ebfc58af70

[4285 rows x 7 columns]

In [79]:
# check to see if we have any missing values
print(clean_df.isnull().sum())

PRODUCT            0
VALUE              0
DATETIME           0
DW_A_NUMBER_INT    0
DW_B_NUMBER_INT    0
CELL_ID            0
SITE_ID            0
dtype: int64


**4.1 TELECOM 1 Data Cleanup**

In [80]:
# start will telecom data set 1
# find all null values
print(df1.isnull().sum())

PRODUCT           0
VALUE             0
DATE_TIME         0
CELL_ON_SITE      0
DW_A_NUMBER       0
DW_B_NUMBER       0
COUNTRY_A         0
COUNTRY_B         0
CELL_ID           0
SITE_ID         676
dtype: int64


In [81]:
# drop the column CELL_ON_SITE,
df1.drop('CELL_ON_SITE', axis=1, inplace=True)
df1

# Drop the column COUNTRY_A
df1.drop('COUNTRY_A', axis=1, inplace=True)
df1

# Drop the column COUNTRY_B
df1.drop('COUNTRY_B', axis=1, inplace=True)
df1

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0        sms      0  ...  +854AcBQT2  +854AcBQT2
1      Voice      0  ...  +854AcBQT2  +854AcBQT2
2        sms      0  ...  +laSrk7g6q  +laSrk7g6q
3      Voice     37  ...  /xvRrCVKoQ  /xvRrCVKoQ
4        sms     25  ...  +bO+qkSonO  +bO+qkSonO
...      ...    ...  ...         ...         ...
4996     sms      0  ...  896459aace  b7d00e59e4
4997     sms     25  ...  7e90b35166  211015b14d
4998     sms      0  ...  fe70e431df  c7ea989264
4999    data      0  ...  ffa6759bb2         NaN
5000   Voice      0  ...  8630058aca  b47a998a3e

[5001 rows x 7 columns]

In [82]:
# data clean up 
# drop NA missing values 
clean_df1 = df1.dropna()
clean_df1

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0        sms      0  ...  +854AcBQT2  +854AcBQT2
1      Voice      0  ...  +854AcBQT2  +854AcBQT2
2        sms      0  ...  +laSrk7g6q  +laSrk7g6q
3      Voice     37  ...  /xvRrCVKoQ  /xvRrCVKoQ
4        sms     25  ...  +bO+qkSonO  +bO+qkSonO
...      ...    ...  ...         ...         ...
4995     sms     25  ...  0e8763bdaa  eefd93f0b4
4996     sms      0  ...  896459aace  b7d00e59e4
4997     sms     25  ...  7e90b35166  211015b14d
4998     sms      0  ...  fe70e431df  c7ea989264
5000   Voice      0  ...  8630058aca  b47a998a3e

[4325 rows x 7 columns]

In [83]:
# check if we still have NA missing values
print(clean_df1.isnull().sum())

PRODUCT        0
VALUE          0
DATE_TIME      0
DW_A_NUMBER    0
DW_B_NUMBER    0
CELL_ID        0
SITE_ID        0
dtype: int64


**4.2 TELECOM 2 Data Cleanup**

In [125]:
# check for NA missing values
print(df2.isnull().sum())

PRODUCT              0
VALUE                0
DATE_TIME            0
CELL_ON_SITE         0
DW_A_NUMBER_INT      0
DW_B_NUMBER_INT      0
COUNTRY_A            0
COUNTRY_B            0
CELLID               0
SIET_ID            605
dtype: int64


In [126]:
# rename the column SIET_ID to SITE_ID
df2.columns = ['PRODUCT', 'VALUE', 'DATETIME', 'CELL_ON_SITE', 'DW_A_NUMBER_INT', 'DW_B_NUMBER_INT', 'COUNTRY_A', 'COUNTRY_B', 'CELL_ID', 'SITE_ID']
df2.head()

PRODUCT  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[5 rows x 10 columns]

In [127]:
# drop the column CELL_ON_SITE,
df2.drop('CELL_ON_SITE', axis=1, inplace=True)
df2

# Drop the column COUNTRY_A
df2.drop('COUNTRY_A', axis=1, inplace=True)
df2

# Drop the column COUNTRY_B
df2.drop('COUNTRY_B', axis=1, inplace=True)
df2

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0      Voice     61  ...  +38u2u/rfx  +38u2u/rfx
1        sms      0  ...  +Fs4mTvdKx  +Fs4mTvdKx
2      Voice     10  ...  +ZaaZozy/+  +ZaaZozy/+
3        sms      0  ...  +cNeJzsTp3  +cNeJzsTp3
4      Voice     22  ...  /eeMklgyJA  /eeMklgyJA
...      ...    ...  ...         ...         ...
4996   Voice      0  ...  2aa04dcab7  f7d63a29e5
4997   Voice     62  ...  7f1f2f889f  b4ddb9e93c
4998   Voice     32  ...  a4731e780e  d68260ef50
4999   Voice     30  ...  98ed726da2  b4130b54ab
5000   Voice     60  ...  1c11a2fcbb  6d70dae094

[5001 rows x 7 columns]

In [128]:
# data clean up 
# drop NA missing values 
clean_df2 = df2.dropna()
clean_df2

PRODUCT  VALUE  ...     CELL_ID     SITE_ID
0      Voice     61  ...  +38u2u/rfx  +38u2u/rfx
1        sms      0  ...  +Fs4mTvdKx  +Fs4mTvdKx
2      Voice     10  ...  +ZaaZozy/+  +ZaaZozy/+
3        sms      0  ...  +cNeJzsTp3  +cNeJzsTp3
4      Voice     22  ...  /eeMklgyJA  /eeMklgyJA
...      ...    ...  ...         ...         ...
4996   Voice      0  ...  2aa04dcab7  f7d63a29e5
4997   Voice     62  ...  7f1f2f889f  b4ddb9e93c
4998   Voice     32  ...  a4731e780e  d68260ef50
4999   Voice     30  ...  98ed726da2  b4130b54ab
5000   Voice     60  ...  1c11a2fcbb  6d70dae094

[4396 rows x 7 columns]

In [129]:
# check to see if we still have NA missing values
print(clean_df2.isnull().sum())

PRODUCT            0
VALUE              0
DATETIME           0
DW_A_NUMBER_INT    0
DW_B_NUMBER_INT    0
CELL_ID            0
SITE_ID            0
dtype: int64


4.3 Cells_geo csv column rename


In [93]:
# drop the 0 column
#df3.drop('Unnamed: 0', axis=1, inplace=True)
#df3

# rename the columns in cells_geo dataset
df3.columns = ['CITY', 'STATUS', 'LOCALISATION', 'DECOUPZONE',	'ZONENAME',	'LONGITUDE',	'LATITUDE',	'REGION',	'AREA',	'CELL_ID', 'SITE_CODE']
df3.head()


CITY      STATUS LOCALISATION  ... AREA     CELL_ID   SITE_CODE
0   ADJAME  In Service      ABIDJAN  ...  NaN  fbb2e75af3  0bb492ef17
1   ADJAME  In Service      ABIDJAN  ...  NaN  1502501d78  0bb492ef17
2   ADJAME  In Service      ABIDJAN  ...  NaN  0c2e182a0c  0bb492ef17
3  ASSINIE  In Service    INTERIEUR  ...  NaN  69bdfc69fd  8a1e6ce31f
4  ASSINIE  In Service    INTERIEUR  ...  NaN  7a1029d2ab  8a1e6ce31f

[5 rows x 11 columns]

In [97]:
# check if there are any NA missing values
print(df3.isnull().sum())

CITY             0
STATUS          67
LOCALISATION     0
DECOUPZONE       0
ZONENAME         6
LONGITUDE        0
LATITUDE         0
REGION           0
AREA            23
CELL_ID          0
SITE_CODE        0
dtype: int64


In [99]:
# drop the the cells with missing values
clean_df3 = df3.dropna()
clean_df3.head()

CITY      STATUS LOCALISATION  ...           AREA     CELL_ID   SITE_CODE
23  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  f740406877  0f34c36ac5
24  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  959e54c8f7  0f34c36ac5
25  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  56b7e7000c  0f34c36ac5
26  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  bc51c17a6f  0f34c36ac5
27  ABOBO  In Service      ABIDJAN  ...  ABIDJAN NORTH  e460975c9f  0f34c36ac5

[5 rows x 11 columns]

In [100]:
# check if there are any cells with no values
print(clean_df3.isnull().sum())

CITY            0
STATUS          0
LOCALISATION    0
DECOUPZONE      0
ZONENAME        0
LONGITUDE       0
LATITUDE        0
REGION          0
AREA            0
CELL_ID         0
SITE_CODE       0
dtype: int64


**5. MERGING**

In [190]:
all_clean = pd.concat([clean_df, clean_df1, clean_df2])
all_clean

PRODUCT  VALUE               DATETIME  ... DATE_TIME DW_A_NUMBER DW_B_NUMBER
5       data      0  2012-05-06 23:07:26.0  ...       NaN         NaN         NaN
20      data      0  2012-05-06 23:10:30.0  ...       NaN         NaN         NaN
23      data      0  2012-05-06 23:11:27.0  ...       NaN         NaN         NaN
25      data      0  2012-05-06 23:12:07.0  ...       NaN         NaN         NaN
33      data      0  2012-05-06 23:13:05.0  ...       NaN         NaN         NaN
...      ...    ...                    ...  ...       ...         ...         ...
4996   Voice      0  2012-05-09 00:01:43.0  ...       NaN         NaN         NaN
4997   Voice     62  2012-05-09 00:01:43.0  ...       NaN         NaN         NaN
4998   Voice     32  2012-05-09 00:01:43.0  ...       NaN         NaN         NaN
4999   Voice     30  2012-05-09 00:01:43.0  ...       NaN         NaN         NaN
5000   Voice     60  2012-05-09 00:01:43.0  ...       NaN         NaN         NaN

[8813 rows x 10 columns]

**6. ANALYSIS**

In [101]:
clean_df3['STATUS'] = clean_df3['STATUS'].str.upper() 
clean_df3.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


CITY      STATUS LOCALISATION  ...           AREA     CELL_ID   SITE_CODE
23  ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  f740406877  0f34c36ac5
24  ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  959e54c8f7  0f34c36ac5
25  ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  56b7e7000c  0f34c36ac5
26  ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  bc51c17a6f  0f34c36ac5
27  ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  e460975c9f  0f34c36ac5

[5 rows x 11 columns]

In [107]:
# find all cell ids that are active
a = 'IN SERVICE'
cities = clean_df3['STATUS'] == a
clean_df3[cities]

CITY      STATUS LOCALISATION  ...           AREA     CELL_ID   SITE_CODE
23       ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  f740406877  0f34c36ac5
24       ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  959e54c8f7  0f34c36ac5
25       ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  56b7e7000c  0f34c36ac5
26       ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  bc51c17a6f  0f34c36ac5
27       ABOBO  IN SERVICE      ABIDJAN  ...  ABIDJAN NORTH  e460975c9f  0f34c36ac5
...        ...         ...          ...  ...            ...         ...         ...
3969  ZUENOULA  IN SERVICE    INTERIEUR  ...   CENTRE NORTH  1cfc2d7ab1  f91ed8d2e9
3970  ZUENOULA  IN SERVICE    INTERIEUR  ...   CENTRE NORTH  3ae1708153  f91ed8d2e9
3971  ZUENOULA  IN SERVICE    INTERIEUR  ...   CENTRE NORTH  cb71f8a29c  61bee9c70c
3972  ZUENOULA  IN SERVICE    INTERIEUR  ...   CENTRE NORTH  d4a6a0b645  61bee9c70c
3973  ZUENOULA  IN SERVICE    INTERIEUR  ...   CENTRE NORTH  f9f3e42589  61bee9c70c

[3880 rows x 11 columns]

We note that 3880 cell ids are active

In [189]:
# TELECOM ANALYSIS
# sum of the amount voice data and sms used for all the days
group_telco1 = all_clean['VALUE'].groupby(all_c['PRODUCT'])
group_telco1.sum()

PRODUCT
Voice    190824
data       2001
sms       39954
Name: VALUE, dtype: int64

In [192]:
# highest voice data used
b = all_clean['PRODUCT'] == 'Voice'
c = all_clean['VALUE'] == all_clean['VALUE'].max()
all_clean[b & c]

PRODUCT  VALUE               DATETIME  ... DATE_TIME DW_A_NUMBER DW_B_NUMBER
1836   Voice   6750  2012-05-09 00:00:23.0  ...       NaN         NaN         NaN

[1 rows x 10 columns]

In [201]:
# highest sms data used
b = all_clean['PRODUCT'] == 'sms'
new_b = all_clean[b]
c = new_b['VALUE'] == new_b['VALUE'].max()
new_b[c].count()

PRODUCT            22
VALUE              22
DATETIME            7
DW_A_NUMBER_INT     7
DW_B_NUMBER_INT     7
CELL_ID            22
SITE_ID            22
DATE_TIME          15
DW_A_NUMBER        15
DW_B_NUMBER        15
dtype: int64

In [209]:
# highest data used
b = all_clean['PRODUCT'] == 'data'
new_b = all_clean[b]
c = new_b['VALUE'] == new_b['VALUE'].max()
new_b[c]

PRODUCT  VALUE  ...      DW_A_NUMBER      DW_B_NUMBER
445    data    748  ...  52fa49c1accc61b  5a4dc9e82deb309

[1 rows x 10 columns]

get the cities
New Section

In [216]:
DF1 = all_clean[['CELL_ID', 'PRODUCT', 'VALUE']]
DF1

CELL_ID PRODUCT  VALUE
5     /lrUXVGVcN    data      0
20    /vLAMOS+Qm    data      0
23    2yuhUrSXQu    data      0
25    2FzOAUm82C    data      0
33    +pZNtCvhmT    data      0
...          ...     ...    ...
4996  2aa04dcab7   Voice      0
4997  7f1f2f889f   Voice     62
4998  a4731e780e   Voice     32
4999  98ed726da2   Voice     30
5000  1c11a2fcbb   Voice     60

[8813 rows x 3 columns]

In [225]:
DF = clean_df3[['CITY', 'CELL_ID']]
DF
a = DF.merge(DF1, how="left")
groupC = a['VALUE'].groupby(a['CITY'])
groupC.max()

CITY
ABENGOUROU      NaN
ABOBO           NaN
ABOISSO         NaN
ABOISSO COMOE   NaN
ABONGOUA        NaN
                 ..
ZATTA           NaN
ZIKISSO         NaN
ZON-HOUNIEN     NaN
ZOUKOUGBEU      NaN
ZUENOULA        NaN
Name: VALUE, Length: 501, dtype: float64